growth-program/src/rewards/deposit.rs

```diff
@@ -35,6 +40,7 @@ pub trait DepositRewardsModule:
         start_week: Week,
         end_week: Week,
         initial_rewards_dollar_per_energy: BigUint,
+        signer: ManagedAddress,
```

Comment:
Done in latest commit.

Reply:
Looks good. I'm wondering if there is any particular reason why you read the payment inside each endpoint and then send it as a parameter in the common function, rather than instantiating it only once inside the common function.

Reply:
Oversight. Fixed.